tests/unit/query/createFilter.ts

```diff
@@ -267,7 +267,7 @@ registerSuite({
 				value: 5,
 				path: createJsonPointer('key', 'key2')
 			};
-			const pickFirstItem = [
+			const pickFirstItem: FilterArray<NestedObj> = [
```

Comment:
Are we ok with NestedObj and SimpleObj as names, or should we spell out the word Object in both?